frontend/src/components/Workspace/Visualize/Plot/ImagePlot.tsx

```diff
@@ -554,34 +696,66 @@ const ImagePlotChart = memo(function ImagePlotChart({
   }
 
   const renderActionRoi = () => {
-    if (!roiDataState?.length || outputKey !== "cell_roi") return null
-    if (!isAddRoi) {
-      return <LinkDiv onClick={addRoi}>Add ROI</LinkDiv>
+    if (!roiFilePath || !roiFilePath.includes(CELL_ROI)) return null
+    if (action) {
+      return (
+        <>
+          {action !== ADD_ROI ? (
+            <BoxDiv>
+              ROI Selecteds: [{pointClick.map((item) => item.z).join(",")}]
+            </BoxDiv>
+          ) : null}
+          <BoxDiv sx={{ display: "flex", gap: 1 }}>
+            <LinkDiv
+              sx={{
+                color:
+                  action === DELETE_ROI
+                    ? "#F84E1B"
+                    : action === MERGE_ROI
+                    ? "#6619A9"
+                    : "default",
+                display: "flex",
+                gap: 1,
+                textDecoration: "none",
+                cursor: "default",
+              }}
+            >
+              {`[${action}]`}
+            </LinkDiv>
+            <LinkDiv onClick={onCancelAdd}>Cancel</LinkDiv>
+            <LinkDiv
+              style={{
+                opacity:
+                  (pointClick.length < 2 && action === MERGE_ROI) ||
+                  (pointClick.length < 1 && action === DELETE_ROI)
+                    ? 0.5
+                    : 1,
+                cursor:
+                  (pointClick.length < 2 && action === MERGE_ROI) ||
+                  (pointClick.length < 1 && action === DELETE_ROI)
+                    ? "default"
+                    : "pointer",
+              }}
+              onClick={addOrSelectRoi}
+            >
+              OK
+            </LinkDiv>
+          </BoxDiv>
+        </>
+      )
     }
-    return (
-      <BoxDiv>
-        <LinkDiv
-          style={{
-            opacity: loadingApi ? 0.5 : 1,
-            cursor: loadingApi ? "progress" : "pointer",
-          }}
-          onClick={addRoiSubmit}
-        >
-          OK
-        </LinkDiv>
-        <LinkDiv
-          style={{
-            opacity: loadingApi ? 0.5 : 1,
-            cursor: loadingApi ? "progress" : "pointer",
-          }}
-          onClick={onCancelAdd}
-        >
-          Cancel
+
+    if (!edit)
+      return (
+        <LinkDiv sx={{ width: "fit-content" }} onClick={editRoi}>
+          Edit ROI
```

Comment:
@itutu-tienday As far as I can see, it seems that in the develop-main request Url has not added the workspace_id param, so that's why it happens

Reply:
@sanglevinh 
Thank you.

> request Url has not added the workspace_id param

Which request is this?
Since the pkl file doesn't seem to be read well by BE (I get a 400 error), is this a request that affects the generation of the pkl file?

Reply:
@sanglevinh 
Even without calling add_roi, the following operation seems to generate a 400 error.

Example)
1. RUN Workflow in develop-main
2. Switch branch to add_ROI_edit+improve-edit-roi
3. Visualize→Select Roi→select cell_roi
4. Select Edit ROI and 400 erro
   - The API for the error is `POST /outputs/image/*/status?workspace_id=1`.

If you don't mind, could you please share the above with BE's?

@ReiHashimoto 
Can you reproduce it in your environment?